src/rascal/structure_managers/structure_manager.hh

```diff
@@ -294,7 +294,7 @@ namespace rascal {
     size_t size() const { return this->implementation().get_size(); }
 
     template <size_t Order, size_t Layer>
-    size_t is_center_atom(const ClusterRefKey<Order, Layer> & cluster) const {
+    bool is_center_atom(const ClusterRefKey<Order, Layer> & cluster) const {
       return (cluster.get_atom_tag() < static_cast<int>(this->size()));
```

Comment:
I can't really comment on the implementation, that should be someone else more familiar with SOAP.

But just quickly here: This only works if you use the `StructureManagerCenters`, where you assign you own tags. If lammps supplies you with its tags , there is not guarantee. Rather use the cluster index of lowest layer and compare it to the size, that should be bullet prove. E.g. you could have atoms tags which are [223, 230002, 4224422]. And size is 3.

Reply:
you are right. I made the modification but now it is only valid for clusters of order 1 and 2.